src/hotspot/cpu/x86/vtableStubs_x86_32.cpp

```diff
@@ -198,35 +200,23 @@ VtableStub* VtableStubs::create_itable_stub(int itable_index) {
   __ load_klass(recv_klass_reg, rcx, noreg);
 
   start_pc = __ pc();
-
-  // Receiver subtype check against REFC.
-  // Destroys recv_klass_reg value.
-  __ lookup_interface_method(// inputs: rec. class, interface
-                             recv_klass_reg, resolved_klass_reg, noreg,
-                             // outputs:  scan temp. reg1, scan temp. reg2
-                             recv_klass_reg, temp_reg,
-                             L_no_such_interface,
-                             /*return_method=*/false);
-
-  const ptrdiff_t  typecheckSize = __ pc() - start_pc;
-  start_pc = __ pc();
-
-  // Get selected method from declaring class and itable index
-  const Register method = rbx;
-  __ load_klass(recv_klass_reg, rcx, noreg); // restore recv_klass_reg
-  __ lookup_interface_method(// inputs: rec. class, interface, itable index
-                             recv_klass_reg, holder_klass_reg, itable_index,
-                             // outputs: method, scan temp. reg
-                             method, temp_reg,
-                             L_no_such_interface);
-
+  __ push(temp_reg);
```

Comment:
Why do we need to save this one? Do we care if this "tmp" is clobbered?

Reply:
This one is still not addressed, in case you missed it, @bulasevich.

Reply:
Right. Thanks.
On x86_32 we have register pressure. I have to push-pop rdx to avoid crash. The question must be caused by temp_reg name, which is similar to rscatch. On x86_32 we do not have any scratch register.
Will it be better if I replace push/pop(temp_reg) with explicit push/pop(rdx)?

Reply:
On x86_32 we do not have any scratch register. Will it be better if I replace push/pop(temp_reg) with explicit push/pop(rdx)?

Yes.